src/snowflake/snowpark/row.py

```diff
@@ -48,19 +59,42 @@ class Row(tuple):
     >>> emp2
     Row(name='James', salary=20000)
 
+    We can optionally make accessing ``Row`` field attributes case insensitive:
+
+    >>> Employee = Row("FIRSTNAME", "SALARY", Row.CASE_INSENSITIVE)
+    >>> emp1 = Employee("John", 10000)
+    >>> emp1.firstName
+    'John'
+    >>> emp2 = Employee("James", 20000)
+    >>> emp2.Salary
+    20000
+
     """
 
+    CASE_INSENSITIVE = "@@@@@@_CASE_INSENSITIVE_@@@@@@"
```

Comment:
Why do we choose to use a constant that might conflict with user data, not a keyword argument?

Reply:
A new `Row` object can be created by either giving `args` or `named_args` but not both. We have a check that throws `ValueError` if both are provided. I can work on this logic to only accept `args` and `named_args` when `len(named_args) == 1` and `"case_sensitive" in named_args`.

Reply:
One caveat is that we will no longer accept `case_sensitive` as an input data when user tries to create a `Row` using `kwargs`. Not sure how much we need to worry about this, but might be a legit concern.

Another way is to create a `Factory` class that can create both case sensitive and insensitive rows, and we could make them share as much code as possible.

Reply:
Not really. If you check the implementation, we accept `args` and `kwargs` only when `len(kwargs) == 1` and `"case_sensitive" in kwargs`. Otherwise we throw error as before.

Let me checkout the `Factory` approach. Sounds much cleaner to me.

Reply:
For the normal `args` case it's okay. But if user provides pure `kwargs` and indeed need data with keyword `case_sensitive`, then they are kinda screwed, right?

Reply:
aah. I see. You are right. In current implementation, when user is passing `kwargs`, we always fix fields to be case sensitive but this is not documented so can confuse the users.